engine/runtime/src/main/scala/org/enso/compiler/pass/analyse/AmbiguousImportsAnalysis.scala

```diff
@@ -0,0 +1,430 @@
+            } else {
+              Right(imp)
+            }
+
+          case None =>
+            Right(moduleImport)
+        }
+
+      // Import all symbols
+      case moduleImport @ IR.Module.Scope.Import.Module(
+            _,
+            _,
+            true,
+            _,
+            hiddenNames,
+            _,
+            false,
+            _,
+            _
+          ) =>
+        getImportTarget(moduleImport, bindingMap) match {
+          case Some(importTarget) =>
+            // Names of the symbols that are exported by a module or a type referred to via importTarget
+            val exportedSymbolNames: List[String] =
+              importTarget.exportedSymbols.keySet.toList
+            val symbolsToIterate = hiddenNames match {
+              case None => exportedSymbolNames
+              case Some(hiddenNamesLiterals) =>
+                val hiddenNames = hiddenNamesLiterals.map(_.name)
+                exportedSymbolNames.filterNot(hiddenNames.contains)
+            }
+            val encounteredErrors: ListBuffer[IR.Error.ImportExport] =
+              ListBuffer()
+            val imp =
+              symbolsToIterate.foldLeft(moduleImport: IR.Module.Scope.Import) {
+                case (imp, symbolName) =>
+                  importTarget.resolveExportedSymbol(symbolName) match {
+                    case Left(resolutionError) =>
+                      throw new CompilerError(
+                        s"Unreachable: (should have been resolved in previous passes) $resolutionError"
+                      )
+                    case Right(resolvedName) =>
+                      tryAddEncounteredSymbol(
+                        module,
+                        encounteredSymbols,
+                        imp,
+                        symbolName,
+                        resolvedName.qualifiedName.toString
+                      ) match {
+                        case Left(error) =>
+                          encounteredErrors += error
+                          imp
+                        case Right(imp) => imp
+                      }
+                  }
+              }
+            if (encounteredErrors.nonEmpty) {
+              Left(encounteredErrors.toList)
+            } else {
+              Right(imp)
+            }
+
+          case None =>
+            Right(moduleImport)
+        }
+
+      // Import a renamed symbol
+      case moduleImport @ IR.Module.Scope.Import.Module(
+            importPath,
+            Some(rename),
+            _,
+            _,
+            _,
+            _,
+            false,
+            _,
+            _
+          ) =>
+        val symbolPath = importPath.name
+        tryAddEncounteredSymbol(
+          module,
+          encounteredSymbols,
+          moduleImport,
+          rename.name,
+          symbolPath
+        ) match {
+          case Left(error) => Left(List(error))
+          case Right(imp)  => Right(imp)
+        }
```

Comment:
Similar comment as before

Reply:
I am confused. Do you mean the one with `fold` suggestion? How should I refactor this to `fold`?